ompi/mca/osc/monitoring/Makefile.am

```diff
@@ -32,8 +32,14 @@ mcacomponent_LTLIBRARIES = $(component_install)
 mca_osc_monitoring_la_SOURCES = $(monitoring_sources)
 mca_osc_monitoring_la_LDFLAGS = -module -avoid-version
 mca_osc_monitoring_la_LIBADD = $(top_builddir)/ompi/lib@OMPI_LIBMPI_NAME@.la \
-	$(OMPI_TOP_BUILDDIR)/ompi/mca/common/monitoring/libmca_common_monitoring.la
+		$(OMPI_TOP_BUILDDIR)/ompi/mca/common/monitoring/libmca_common_monitoring.la
 
 noinst_LTLIBRARIES = $(component_noinst)
 libmca_osc_monitoring_la_SOURCES = $(monitoring_sources)
 libmca_osc_monitoring_la_LDFLAGS = -module -avoid-version
+
+# C file depends on the header file generated at configure time
+$(srcdir)/osc_monitoring_component.c: $(builddir)/osc_monitoring_template_gen.h
```

Comment:
What's the point of this rule?  You don't have any generating rules, so all it's doing is expressing a dependency, which should be picked up in dependency generation.  It's not harmful, but it's odd.

Reply:
Expressing the dependency was the only point. I wasn't sure whether or not it would have been expressed otherwise. I think I tried to check that if the generated .h file was edited by hand (should not be, but who knows...) osc_monitoring_component.c was compiled again, and I think it was not. If it should be here, I'll remove it.

Reply:
you can look at the .deps/...  file to make sure, but the compiler should generate a dependency line.  In general we don't add extra dependency information unless we're adding a build rule.  So it's not required, but it's also not harmful.  If you can change it, that would be great.  But I'm also fine if you change it in a follow-up commit.